### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by Klarstein**
## Description:
Change initial timeout after startup from 3s to 2m in order to make the first run pass the health-check. 

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ ENV USERNAME="" PASSWORD=""
 EXPOSE 6052
 
 # Run healthcheck (heartbeat)
-HEALTHCHECK --interval=5m --timeout=3s \
+HEALTHCHECK --interval=5m --timeout=2m \
```
Comment (rjmidau) [inline docker/Dockerfile:19]: https://docs.docker.com/engine/reference/builder/#healthcheck
> The health check will first run **interval** seconds after the container is started, and then again **interval** seconds after each previous check completes.
If a single run of the check takes longer than **timeout** seconds then the check is considered to have failed.
It takes **retries** consecutive failures of the health check for the container to be considered `unhealthy`.
**start period** provides initialization time for containers that need time to bootstrap. Probe failure during that period will not be counted towards the maximum number of retries. 

The values we're using mean the first health check would not run until 5m after the container starts, and cURL would be allowed 2 minutes to complete. I suggest these interval and timeout values are still too high, and should be set to something closer to the defaults (interval and timeout at 30s each).

Until the container is marked as healthy, services like Traefik will not allow traffic to be routed to the container.

If we're worried about the container taking longer than 90 seconds to start (exceeding the default 3 retries) and being marked unhealthy, I recommend setting the start-period (perhaps to 30 seconds also), although I've never encountered esphome taking an excessively long to start.

Comment (Klarstein) [reply docker/Dockerfile:19]: I completely agree, the proposed change was a response to https://github.com/esphome/esphome/pull/1492#issuecomment-774487297

Comment (rjmidau) [reply docker/Dockerfile:19]: That's awesome! Can you update your PR to use lower interval and timeout values (ie. 30s)?

**Review by rjmidau — APPROVED** [review]
Looks Great!

**Review by jesserockz — APPROVED** [review]

**Pull request merged**